server/routes/appointments/create-and-edit/handlers/location.test.ts

```diff
@@ -91,12 +96,48 @@ describe('Route Handlers - Create Appointment - Location', () => {
 
       when(prisonService.getLocationsForAppointments).mockResolvedValue(locations)
 
-      await handler.POST(req, res)
+      await handler.CREATE(req, res)
 
       expect(res.validationFailed).toHaveBeenCalledWith('locationId', `Selected location not found`)
     })
   })
 
+  describe('EDIT', () => {
+    it('should update the occurrence and redirect back to the occurrence details page', async () => {
+      req.body = {
+        locationId: 26152,
+      }
+
+      req.params = {
+        appointmentId: '2',
+        occurrenceId: '12',
+      }
+
+      req.session.appointmentJourney = {
+        location: {
+          id: 26152,
+          description: 'Chapel',
+        },
+      } as unknown as AppointmentJourney
+
+      when(prisonService.getLocationsForAppointments).mockResolvedValue(locations)
+      when(activitiesService.editAppointmentOccurrence).calledWith(atLeast(12))
+
+      await handler.EDIT(req, res)
+
+      expect(activitiesService.editAppointmentOccurrence)
+
+      expect(req.flash).toHaveBeenCalledWith(
+        'successMessage',
+        JSON.stringify({
+          message: `Appointment location changed successfully`,
+        }),
+      )
+
+      expect(res.redirectOrReturn).toHaveBeenCalledWith('/appointments/2/occurrence/12')
+    })
+  })
+
```

Comment:
Perhaps add a test for the edit handler that tests location id is validated? Realise it's a shared method between create and edit but a test would confirm that the validation logic isn't accidentally removed